### PR TITLE
Add negative tests for context API functions

### PR DESCRIPTION
**Pull request by MichaelRizkalla-arm**
This PR adds negative tests for the following API functions:

- clCreateContext
- clCreateContextFromType
- clRetainContext
- clReleaseContext
- clSetContextDestructorCallback

Also, define a new macro `test_object_failure_ret` to test for an expected error code and the returned object is `NULL` as a result of a failure.

### CODE REVIEW

File: test_conformance/api/negative_context.cpp [hunk 1]
```diff
@@ -0,0 +1,300 @@
...
+        props[1] = reinterpret_cast<cl_context_properties>(nullptr);
+        ctx = clCreateContext(props, 1, &device, nullptr, nullptr, &err);
+        test_object_failure_ret(
+            ctx, err, CL_INVALID_PROPERTY,
+            "clCreateContext should return CL_INVALID_PROPERTY when: \"context "
+            "property name in properties is not a supported property name\"",
+            TEST_FAIL);
+
+        cl_context_properties invalid_value{ -1 };
+        props[0] = CL_CONTEXT_INTEROP_USER_SYNC;
```
Comment (EwanC) [inline test_conformance/api/negative_context.cpp:46]: we could enter this control flow if `get_device_cl_version(device)` returns `Version(1,1)` but `CL_CONTEXT_INTEROP_USER_SYNC` isn't added to the spec until version 1.2

File: test_conformance/api/negative_context.cpp [hunk 2]
```diff
@@ -0,0 +1,300 @@
...
+
+    cl_context_info INVALID_PARAM_VALUE = 0;
+    err = clGetContextInfo(context, INVALID_PARAM_VALUE, 0, nullptr, nullptr);
+    test_failure_error_ret(
+        err, CL_INVALID_VALUE,
+        "clGetContextInfo should return CL_INVALID_VALUE when: \"param_name is "
+        "not one of the supported values\"",
+        TEST_FAIL);
+
+    err = clGetContextInfo(context, CL_CONTEXT_REFERENCE_COUNT, 0, &param_value,
```
Comment (EwanC) [inline test_conformance/api/negative_context.cpp]: using a non-zero value, e.g `sizeof(cl_uint) - 1` would be a marginally stronger test.

File: test_conformance/api/negative_context.cpp [hunk 3]
```diff
@@ -0,0 +1,300 @@
...
+            "clCreateContext should return CL_INVALID_PROPERTY when: \"the "
+            "value specified for a supported property name is not valid\"",
+            TEST_FAIL);
+
+        props[0] = CL_DEVICE_TYPE;
+        props[1] = 0;
+        ctx = clCreateContext(props, 1, &device, nullptr, nullptr, &err);
+        test_object_failure_ret(
+            ctx, err, CL_INVALID_PROPERTY,
+            "clCreateContext should return CL_INVALID_PROPERTY when: \"the "
```
Comment (EwanC) [inline test_conformance/api/negative_context.cpp]: Does this provide extra coverage that the first check for this case on line 41 doesn't?

File: test_conformance/api/negative_context.cpp (hunk 1 above)
Comment (MichaelRizkalla-arm) [reply test_conformance/api/negative_context.cpp:46]: I updated the branch condition to `>= Version(1, 2)`.

File: test_conformance/api/negative_context.cpp (hunk 2 above)
Comment (MichaelRizkalla-arm) [reply test_conformance/api/negative_context.cpp]: Done

File: test_conformance/api/negative_context.cpp (hunk 3 above)
Comment (MichaelRizkalla-arm) [reply test_conformance/api/negative_context.cpp]: No, it's been removed.

**Comment by kpet** [comment]
Merging as agreed in 2026/01/13 teleconference.
